crates/kitsune_p2p/kitsune_p2p/tests/sharding.rs

```diff
@@ -0,0 +1,424 @@
+use crate::common::{
+    start_bootstrap, start_signal_srv, wait_for_connected, KitsuneTestHarness, TestHostOp,
+};
+use fixt::fixt;
+use kitsune_p2p::actor::{BroadcastData, KitsuneP2pSender};
+use kitsune_p2p::dht::arq::LocalStorageConfig;
+use kitsune_p2p::dht::prelude::SpaceDimension;
+use kitsune_p2p::dht_arc::DhtLocation;
+use kitsune_p2p_bin_data::fixt::KitsuneSpaceFixturator;
+use kitsune_p2p_bin_data::{KitsuneAgent, KitsuneBasis, KitsuneBinType};
+use kitsune_p2p_fetch::FetchContext;
+use kitsune_p2p_types::config::tuning_params_struct;
+use kitsune_p2p_types::dht::{Arq, ArqStrat};
+use kitsune_p2p_types::KitsuneTimeout;
+use num_traits::AsPrimitive;
+use std::sync::Arc;
+
+mod common;
+
+/// Test scenario steps:
+///   1. Set up 5 nodes, each with one agent.
+///   2. Assign a DHT arc to each agent such that they overlap with the next agent's start location.
+///   3. Connect each agent to the next agent (circular), so that we know they are aware of each other.
+///   4. Publish an op with a basis location set to the location of the 4th agent. This should also be visible to the 3rd agent by 2. above.
+///   5. Wait for the 3rd to receive the data.
+///   6. Assert that the op was never published to the 1st, 2nd, or 5th agents. (Note that we cannot check if we sent it to ourselves because the op was already in our store)
```

Comment:
We should totally do this for all tests, I have tried this in the past but sometimes forget how helpful a synopsis is.

Reply:
For longer tests I find it really useful to have a bit of a guide yeah :)